packages/asset-swapper/src/utils/aggregate.ts

```diff
@@ -0,0 +1,800 @@
+     * small discrepancies between samples and truth.
+     * Default is 0.0005 (5 basis points).
+     */
+    bridgeSlippage: number;
+    /**
+     * Number of samples to take for each DEX quote.
+     */
+    numSamples: number;
+    /**
+     * Dust amount, as a fraction of the fill amount.
+     * Default is 0.01 (100 basis points).
+     */
+    dustThreshold: number;
+}
+
+const IMPROVE_ORDERS_OPTS_DEFAULTS = {
+    runLimit: 1024,
+    excludedSources: [],
+    bridgeSlippage: 0.0005,
+    dustThreshold: 0.01,
+    numSamples: 8,
+    noConflicts: false,
+};
+
+// Used internally by `FillsOptimizer`.
+interface FillsOptimizerContext {
+    currentPath: Fill[];
+    currentPathInput: BigNumber;
+    currentPathOutput: BigNumber;
+    currentPathFlags: number;
+}
+
+/**
+ * Represents an individual DEX sample from the sampler contract.
+ */
+export interface DexSample {
+    source: ERC20BridgeSource;
+    input: BigNumber;
+    output: BigNumber;
+}
+
+/**
+ * Flags for `Fill` objects.
+ */
+export enum FillFlags {
+    SourceNative = 0x1,
+    SourceUniswap = 0x2,
+    SourceEth2Dai = 0x4,
+    SourceKyber = 0x8,
+}
+
+/**
+ * Improve a market sell operation by (potentially) merging native orders with
+ * generated bridge orders.
+ * @param samplerContract The `IERC20BridgeSamplerContract` contract.
+ * @param nativeOrders Native orders.
+ * @param takerAmount Amount of taker asset to sell.
+ * @param opts Options object.
+ * @return Improved orders.
+ */
+export async function improveMarketSellAsync(
+    samplerContract: IERC20BridgeSamplerContract,
+    nativeOrders: SignedOrderWithoutDomain[],
+    takerAmount: BigNumber,
+    opts?: Partial<ImproveOrdersOpts>,
+): Promise<SignedOrderWithoutDomain[]> {
+    if (nativeOrders.length === 0) {
+        throw new Error(AggregationError.EmptyOrders);
+    }
+    const _opts = {
+        ...IMPROVE_ORDERS_OPTS_DEFAULTS,
+        ...opts,
+    };
+    const [fillableAmounts, dexQuotes] = await queryNetworkAsync(
+        samplerContract,
+        nativeOrders,
+        getSampleAmounts(takerAmount, _opts.numSamples),
+        difference(SELL_SOURCES, _opts.excludedSources),
+    );
+    const nativePath = pruneDustFillsFromNativePath(
+        createSellPathFromNativeOrders(nativeOrders, fillableAmounts),
+        takerAmount,
+        _opts.dustThreshold,
+    );
+    const dexPaths = createPathsFromDexQuotes(dexQuotes, _opts.noConflicts);
+    const allPaths = [...dexPaths];
+    const allFills = flattenDexPaths(dexPaths);
+    // If native orders are allowed, splice them in.
+    if (!(_opts.excludedSources as ERC20BridgeSource[]).includes(ERC20BridgeSource.Native)) {
+        allPaths.splice(0, 0, nativePath);
+        allFills.splice(0, 0, ...nativePath);
+    }
+    const optimizer = new FillsOptimizer(_opts.runLimit);
+    const optimalPath = optimizer.optimize(
+        sortFillsByPrice(allFills),
+        takerAmount,
+        pickOptimalPath(allPaths, takerAmount),
```

Comment:
is it possible to move this `pickOptimalPath` logic into `FillsOptimizer`? Seems like it makes more sense there.

Reply:
`pickOptimalPath` just picks the shortest path of those passed in, but `FillsOptimizer`'s job is to actually generate new paths so idk if that's a great fit.